spaceship/lib/spaceship/portal/portal_client.rb

```diff
@@ -465,6 +465,22 @@ def provisioning_profiles(mac: false)
       end
     end
 
+    ##
+    # this endpoint is used by Xcode to fetch provisioning profiles.
+    # The response is an xml plist but has the added benefit of containing the appId of each provisioning profile.
+    def provisioning_profiles_via_xcode_api(mac: false)
+      req = request(:post) do |r|
+        r.url "https://developerservices2.apple.com/services/#{PROTOCOL_VERSION}/#{platform_slug(mac)}/listProvisioningProfiles.action"
+        r.params = {
+          teamId: team_id,
+          includeInactiveProfiles: true,
+          onlyCountLists: true
+        }
+      end
+
+      parse_response(req, 'provisioningProfiles')
```

Comment:
My understanding is that the csrf code was only necessary for modifying provisioning profiles. Since the create/delete/repair requests go to the developer portal domain, we don't need it for those requests. I will double check to make sure my understanding is correct.

I will test more thoroughly